### DISCUSSION TITLE
Is it possible to make closing the app window `Exit Watch Later`?

### DISCUSSION THREAD

**Discussion by parasiteoflife** [mpvnet-player/mpv.net#551] [Q&A] [answered]
I've set the shortkey to Exit Watch Later to `Q`, `Ctrl+w` and `Ctrl+W`, but I often just close the window clicking on the X button rather than pressing the shortkeys and I have been wondering this for a long time.  
I would like to make every exit operation an Exit Watch Later one, so in short I always want to save the Watch Later status no matter how I close the application, can this be changed userwide or would it have to be a feature request?

EDIT: Here is the script I'm using:

```lua
msg = require 'mp.msg'

function shutdown()
    msg.info('saving watch later...')
    mp.command("write-watch-later-config")
end

mp.register_event('shutdown', shutdown)
```

**Comment by stax76** [comment, marked as answer]
This might be possible with a user script. You can try getting help with the code, asking on [reddit](https://www.reddit.com/r/mpv/).

There is a [shutdown](https://mpv.io/manual/master/#command-interface-shutdown) event.

When this event happens, it's possible to execute the input command [write-watch-later-config](https://mpv.io/manual/master/#command-interface-write-watch-later-config).
